src/price-spread-ratio/price-spread-ratio.spec.js

```diff
@@ -0,0 +1,261 @@
+      // set up prices that will not trigger an alert
+      const mockPerpPrice = new BigNumber(101);
+      const mockPerpPriceWei = mockPerpPrice.times(WEI_PER_ETHER);
+
+      // set up the mock contract methods to return the price information
+      data.contractsData[0].contract = {
+        getIndexPrice: jest.fn().mockResolvedValue(mockPerpPriceWei),
+      };
+
+      // create a block event that will initialize the timestamp on the contract data object
+      const blockEvent = {
+        block: {
+          timestamp: 1,
+        },
+      };
+
+      // run the handler
+      let findings = await handleBlock(blockEvent);
+
+      // check assertions
+      expect(findings).toStrictEqual([]);
+
+      // intentionally set the next timestamp to be beyond the time threshold
+      blockEvent.block.timestamp += (1 + Number(data.contractsData[0].timeThresholdSeconds));
+
+      // run the handler again
+      findings = await handleBlock(blockEvent);
+
+      // check assertions
+      // axios.get will not be called for the first run of the handler
+      expect(axios.get).toHaveBeenCalledTimes(1);
+      expect(findings).toStrictEqual([]);
+    });
+
+    it('returns empty findings if the price spread ratio it beyond the threshold but not enough time has elapsed', async () => {
+      // set up prices that will trigger an alert
+      const mockPerpPrice = new BigNumber(102);
+      const mockPerpPriceWei = mockPerpPrice.times(WEI_PER_ETHER);
+
+      // set up the mock contract methods to return the price information
+      data.contractsData[0].contract = {
+        getIndexPrice: jest.fn().mockResolvedValue(mockPerpPriceWei),
+      };
+
+      const blockEvent = {
+        block: {
+          timestamp: 1,
+        },
+      };
+
+      // run the handler
+      let findings = await handleBlock(blockEvent);
+
+      // check assertions
+      expect(findings).toStrictEqual([]);
+
+      // intentionally set the next timestamp to be beyond the time threshold
+      blockEvent.block.timestamp += (Number(data.contractsData[0].timeThresholdSeconds) - 1);
+
+      // run the handler again
+      findings = await handleBlock(blockEvent);
+
+      // check assertions
+      // axios.get will not be called for the first run of the handler
+      expect(axios.get).toHaveBeenCalledTimes(1);
+      expect(findings).toStrictEqual([]);
+    });
+
+    it('returns findings if the price spread ratio is too high', async () => {
+      // set up prices that will trigger an alert
+      const mockPerpPrice = new BigNumber(102);
+      const mockPerpPriceWei = mockPerpPrice.times(WEI_PER_ETHER);
+
+      // set up the mock contract methods to return the price information
+      data.contractsData[0].contract = {
+        getIndexPrice: jest.fn().mockResolvedValue(mockPerpPriceWei),
+      };
+
+      const blockEvent = {
+        block: {
+          timestamp: 1,
+        },
+      };
```

Comment:
pretty sure all this can be moved to the `beforeEach` and expose the things you need to change with `let`s outside of the `beforeEach`s scope, like `mockPerpPrice` and `blockEvent`. I _think_ it should work. also to make it easy if it does, make a function that updates both `mockPerpPrice` and `mockPerpPriceWei` so it's ezpz 1 line updating

Reply:
YOu can also just not do this and I wouldn't mind